content/client-lib-development-guide/features.textile

```diff
@@ -232,6 +232,22 @@ h3(#rest-auth). Auth
 ** @(RSA16b)@ If the library is provided with a token without the corresponding @TokenDetails@, then this holds a @TokenDetails@ instance in which only the @token@ attribute is populated with that token string
 ** @(RSA16c)@ Is set with the current token (if applicable) on instantiation and each time it is replaced, whether the result of an explicit @Auth#authorize@ operation, or a library-initiated renewal resulting from expiry or a token error response
 ** @(RSA16d)@ Is empty if there is no current token, including after a previous token has been determined to be invalid or expired, or if the library is using basic auth
+* @(RSA17)@ @Auth#revokeTokens@ function: This function is intended to be used to revoke tokens using provided @TokenRevocationRequest@ details. It internally creates request for  "REST revokeTokens endpoint":/rest-api/#revokeTokens and handles the response.
+** @(RSA17a)@ Method signature is @revokeTokens(TokenRevocationRequest)@.
+** @(RSA17b)@ If the @TokenRevocationRequest#key@ attribute is not provided, then the function must throw an error with a @"Key is not provided"@ message.
+** @(RSA17c)@ If the @TokenRevocationRequest#targets@ attribute is null or invalid (as per @TF3@ ), then the function must throw an error with a @"targets are invalid"@ message.
+** @(RSA17d)@ If the @TokenRevocationRequest#key@ is provided and @TokenRevocationRequest#targets@ valid, then the function must create the request and execute it.
+** @(RSA17e)@ Function returns a @TokenRevocationResponse@ object that contains the result of the request if request is successful.
+** @(RSA17f)@ Function throws an appropriate error/exception  if the request fails either for the reasons of invalid request, or as  a result of request failing.
+** @(RSA17g)@ Tests in this section must exist in client library.
+*** @(RSA17g1)@ Given a null @TokenRevocationRequest#key@ attribute, verify that the function throws an error with a @"Key is not provided"@ message.
+*** @(RSA17g2)@ Given a null @TokenRevocationRequest#targets@ attribute, verify that the function throws an error with a @"targets are invalid"@ message.
```

Comment:
Could you not simplify this to `Given a null or empty ...` and remove the next spec item?

Reply:
My intention was to separate the null and empty situations however it looks like the spec does not separate them. I am keen on adding a new spec item for throwing a different error when target is null. WDYT?

Reply:
Added a spec item and split two cases 1ae5367e5358f2e7845785dfed9e0cf9a81e71f9

Reply:
Well if you feel strongly it's needed.  IMHO, it creates "more spec", which I believe we are all not keen on. So whatever we can do to keep the spec as terse as possible, and have single spec items that cover a wider array of problems without being overly explicit, the better.

Reply:
My initial motivation was to be as clear as possible to clients about any input they provide. But I get your point. I have made change bac65329d67ecbd6896a1e0b34dc2ccb0285b55b with also clarifying the message so at least we are more descriptive in message side